### PR TITLE
fix #3331 share with WordPress permissions

### LINKED ISSUE #3331: Permissions check needed when sharing media to the WordPress app
We've got permissions working when a user attempts to add media from within the app, but when sharing from other apps (Photos, for example) it fails without asking for permission.

Reproduce:
* Android 6.0, disable media permissions
* Open Photos (or other app with media to share)
* Share a photo (or multiple) with WordPress, add to Media Library option
* Failure toast, no prompt to enable permissions

### PR DESCRIPTION
**Pull request by maxme**
fix #3331 share with WordPress permissions

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/media/WordPressMediaUtils.java [hunk 1]
```diff
@@ -38,31 +39,37 @@
...
+    public static boolean checkCameraAndStoragePermissions(Activity activity, int requestCode) {
+        return checkMediaPermissions(activity, requestCode, new String[] {
+                Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                Manifest.permission.CAMERA
+        });
+    }
+
+    public static boolean checkStoragePermission(Activity activity, int requestCode) {
+        return checkMediaPermissions(activity, requestCode, new String[] {
+                permission.WRITE_EXTERNAL_STORAGE
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/media/WordPressMediaUtils.java]: Nitpick,  should this be `Manifest.permission.WRITE_EXTERNAL_STORAGE` for consistency  with line 62.

**Comment by maxme** [comment]
@daniloercoli thanks for the review, I moved permission checks and requests to a new utils class `PermissionUtils`

**Comment by daniloercoli** [comment]
I tested this PR on emulator only, but should be safe to be merged. Let me know what you want to do, If I can merge it or we need to wait someone with a real device.

**Comment by maxme** [comment]
Merge it please, I tested on a Nexus 9 with Marshmallow.

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**